### PR TITLE
Cover more test/widgets tests with leak tracking #3

### PR DESCRIPTION
**Pull request by ksokolovskyi**
### Description
This PR changes `testWidgets` by `testWidgetsWithLeakTracking` in number of `flutter/test/widgets` tests according to the https://github.com/dart-lang/leak_tracker/issues/134.

### Marks leaks
- https://github.com/flutter/flutter/issues/134572
- https://github.com/flutter/flutter/issues/134575

### Tests
- Updates `form_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `fractionally_sized_box_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `framework_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `gesture_detector_semantics_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `gesture_detector_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `gesture_disambiguation_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `global_keys_duplicated_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `global_keys_moving_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `grid_paper_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `grid_view_layout_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `grid_view_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `heroes_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `hit_testing_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `hyperlink_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `icon_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `image_filter_quality_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `image_filter_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `image_headers_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `image_icon_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `image_resolution_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `image_rtl_test.dart` to use `testWidgetsWithLeakTracking`.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

**Comment by ksokolovskyi** [comment]
cc @polina-c

File: packages/flutter/test/widgets/grid_view_test.dart [hunk 1]
```diff
@@ -663,9 +664,18 @@ void main() {
     final TestClipPaintingContext context = TestClipPaintingContext();
     renderObject.paint(context, Offset.zero);
     expect(context.clipBehavior, equals(Clip.none));
-  });
-
-  testWidgets('GridView respects clipBehavior', (WidgetTester tester) async {
+  },
+  leakTrackingTestConfig: const LeakTrackingTestConfig(
+    // TODO(someone): remove after fixing
```
Comment (polina-c) [inline packages/flutter/test/widgets/grid_view_test.dart:669]: If the test fail and the failure seems to be already covered by an issue, just leave it non converted `testWidgets` for now.

Later we will either convert it after issue is fixed or investigate and create separate issue.

File: packages/flutter/test/widgets/grid_view_test.dart [hunk 2]
```diff
@@ -729,9 +739,14 @@ void main() {
     // 4th, check that a non-default clip behavior can be sent to the painting context.
     renderObject.paint(context, Offset.zero);
     expect(context.clipBehavior, equals(Clip.antiAlias));
-  });
-
-  testWidgets('GridView.builder respects clipBehavior', (WidgetTester tester) async {
+  },
+  leakTrackingTestConfig: const LeakTrackingTestConfig(
+    // TODO(someone): remove after fixing
```
Comment (polina-c) [inline packages/flutter/test/widgets/grid_view_test.dart:744]: Replace someone with actual alias.

**Review by polina-c — COMMENTED** [review]
couple small comments

**Comment by ksokolovskyi** [comment]
@polina-c thanks for the review.

**Comment by ksokolovskyi** [comment]
@polina-c, what to do if it happens that there is the same leak in a couple of `testWidgetsWithLeakTracking`?
Do I need to mark the leak in only one of them and revert all others to the `testWidgets,` or do I need to mark all the tests?

**Comment by polina-c** [comment]
> @polina-c, what to do if it happens that there is the same leak in a couple of `testWidgetsWithLeakTracking`? Do I need to mark the leak in only one of them and revert all others to the `testWidgets,` or do I need to mark all the tests?

Thanks for asking. I will add answer to top description of the main issue. Let's convert just one test. It will be enough for issue resolution. When we have too much issues, we need to stop conversion and fix them, before going forward.

**Review by polina-c — APPROVED** [review]
LGTM

**Comment by ksokolovskyi** [comment]
> > @polina-c, what to do if it happens that there is the same leak in a couple of `testWidgetsWithLeakTracking`? Do I need to mark the leak in only one of them and revert all others to the `testWidgets,` or do I need to mark all the tests?
> 
> Thanks for asking. I will add answer to top description of the main issue. Let's convert just one test. It will be enough for issue resolution. When we have too many issues, we need to stop conversion and fix them, before going forward.

Thanks a lot for the answer and review!

**Pull request merged**